elements/lisk-codec/src/varint.ts

```diff
@@ -114,3 +116,41 @@ export const readSInt64 = (buffer: Buffer, offset: number): [bigint, number] =>
 	}
 	return [-(varInt + BigInt(1)) / BigInt(2), size];
 };
+
+const varintMinLength = (n: bigint): number => {
+	if (n < BigInt(1) << BigInt(7 * 1)) {
+		return 1;
+	}
+	if (n < BigInt(1) << BigInt(7 * 2)) {
+		return 2;
+	}
+	if (n < BigInt(1) << BigInt(7 * 3)) {
+		return 3;
+	}
+	if (n < BigInt(1) << BigInt(7 * 4)) {
+		return 4;
+	}
+	if (n < BigInt(1) << BigInt(7 * 5)) {
+		return 5;
+	}
+	if (n < BigInt(1) << BigInt(7 * 6)) {
+		return 6;
+	}
+	if (n < BigInt(1) << BigInt(7 * 7)) {
+		return 7;
+	}
+	if (n < BigInt(1) << BigInt(7 * 8)) {
+		return 8;
+	}
+	if (n < BigInt(1) << BigInt(7 * 9)) {
+		return 9;
+	}
+	return 10;
+};
+
+const validateVarintSize = (result: bigint, size: number) => {
```

Comment:
Maybe it could be possible to be more efficient by 'merging' the checks, i.e. have something like

```typescript
if ( !( (BigInt(1) << BigInt(7 * (size - 1)) <= result < BigInt(1) << BigInt(7 * size) ) ) {
    throw new Error('invalid varint bytes. vartint must be in shortest form.');
}
```

Reply:
I think it would work but result == `0` needs to be handled separately